src/components/Pressable/utils.ts

```diff
@@ -119,6 +119,56 @@ const adaptTouchEvent = (event: GestureTouchEvent): PressableEvent => {
   };
 };
 
+type StylePropKeys = (keyof ViewStyle)[];
+
+// Source:
+// - From ViewStyle extracted FlexStyle sub-interface which contains all of the box-model manipulating props.
+// - From FlexStyle handpicked those styles, which act on the inner part of the box-model.
+const innerStyleKeys = [
+  'alignContent',
+  'alignItems',
+  'flexBasis',
+  'flexDirection',
+  'flexWrap',
+  'rowGap',
+  'gap',
+  'columnGap',
+  'justifyContent',
+  'overflow',
+  'padding',
+  'paddingBottom',
+  'paddingEnd',
+  'paddingHorizontal',
+  'paddingLeft',
+  'paddingRight',
+  'paddingStart',
+  'paddingTop',
+  'paddingVertical',
+  'start',
+  'end',
+  'direction', // iOS only
+] as StylePropKeys;
+
+type StyleTuple = [ViewStyle, ViewStyle];
+const splitStyles = (from: ViewStyle): StyleTuple => {
+  if (!from) {
+    return [{}, {}] as StyleTuple;
+  }
+
+  const outerStyles = { ...from };
+
+  const innerStyles = Object.fromEntries(
+    innerStyleKeys
+      .filter((key) => key in from)
+      .map((key) => {
+        outerStyles[key] = undefined;
+        return [key, from[key]];
+      })
+  ) as ViewStyle;
```

Comment:
I had something more like this in mind:

```jsx
const outerStyles = {}
const innerStyles = {}

for (const key in from) {
  if (innerStyleKeys.has(key)) {
    innerStyles[key] = from[key]
  } else {
    outerStyles[key] = from[key]
  }
}
```

I think it's a bit more concise and avoids the dirty side-effect inside `map`. But it requires `innerStyleKeys` to have O(1) time access, hence set in the other comment.

Reply:
Done, working in: [75d4109](https://github.com/software-mansion/react-native-gesture-handler/pull/2982/commits/75d41093958d014b2a2c7f96cad9a1067b7904fb)